File: src/arrays_and_strings/java/IsUnique.java
package arrays_and_strings.java;

import java.util.Scanner;

/**
 *
 */
public class IsUnique {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    String str = in.next();

    System.out.println(isUnique(str) ? "Yes" : "No");
  }

  private static boolean isUnique(String str) {
    for (int i = 0; i < str.length(); i++) {
      int index = str.indexOf(str.charAt(i), i + 1);
      if (index < 0 || index >= str.length()) return false;
    }
    return true;
  }
}

File: src/arrays_and_strings/kotlin/IsUnique.kt
package arrays_and_strings.kotlin

/**
 *
 */
fun main(args: Array<String>) {
  val str = readLine().toString()

  print(if (isUnique(str)) "Yes" else "No")
}

fun isUnique(str: String): Boolean {
  for (i in str.indices) {
    val index = str.indexOf(str[i], i + 1)
    if (index in str.indices) return false
  }
  return true
}